hbase-client/src/test/java/org/apache/hadoop/hbase/client/TestScan.java

```diff
@@ -192,8 +187,7 @@ public void testSetStartRowAndSetStopRow() {
       scan.withStartRow(new byte[HConstants.MAX_ROW_LENGTH + 1]);
       fail("should've thrown exception");
     } catch (IllegalArgumentException iae) {
-    } catch (Exception e) {
-      fail("expected IllegalArgumentException to be thrown");
+      // Expected
```

Comment:
I think we could file another issue to change these tests to use assertThrows. Anyway, the current modifications make the code better, can open a follow on issue for changing to assertThrows.